app/components/channel_system_message/channel_system_message.js

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import {Posts} from 'mattermost-redux/constants';
+import FormattedMarkdownText from 'app/components/formatted_markdown_text';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {ViewTypes} from 'app/constants';
+
+
+export default class MiscSystemMessage extends React.PureComponent {
+
+    render() {
+        const { post, theme, navigator, textStyles } = this.props;
+        console.log('theme:', theme);
+
+        const getStyleSheet = makeStyleSheetFromTheme((theme) => {
```

Comment:
The function created by `makeStyleSheetFromTheme` is meant to only return a new `StyleSheet` when the `theme` changes, so it needs to be defined outside of the render method for that memoization to work correctly